packages/smooth_app/lib/helpers/attributes_card_helper.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:openfoodfacts/model/Attribute.dart';
+
```

Comment:
I would create a static class: I don't like those methods here and there.

Within that helper class, at least some `static const int _MATCH_THRESHOLD_1 = 20;`.

I would even code it more OOP, that would prevent minor bugs like `attribute.match! <= 20` in one method and `attribute.match! < 20` in the other.
Something like `AttributeClassHelper(attribute).getText()`.

But at least, a static class.

Reply:
hmm good point on keeping things consistent to avoid minor bugs, your comment helped me realize that most of the code is the same among the static methods and there was a great potential for code sharing. Please take another look.

Regarding static class: I'm following the Dart guideline (which actually you referred in a different PR :D )
https://dart.dev/guides/language/effective-dart/design#avoid-defining-a-class-that-contains-only-static-members

Reply:
@monsieurtanuki Thanks a lot for going through the effort of restructuring the class, but to be honest I don't agree with the restructure on 3 counts.

1. It's a cognitive load, specially the "_computeThresholdIndex", it's not straightforward to understand in a single read.
2. It's easy to forget that the index in one of the constants (like _ICON_CHARACTERS) is imperative to the logic of the class. As it's a lot of hoops to realize how it's used (_ICON_CHARACTERS -> _thresholdIndex -> _computeThresholdIndex)
3. I'm not sure I see an advantage in creating an instance of this class, since everytime one uses this class one will always create a new instance for it.

Also I don't agree that we need to create a class just because all the helper methods are related to a single entity (in this case "Attribute"), that was actually my plan, which is why I created a specific file for encapsulating all the closely related functions, In my opinion any file should only contain strongly related functions, but that should not lead to the argument for turning all such files (with static methods inside) into a class.

That said, I'm open to creating a class wrapper with static methods inside where you pass in the Attribute to each method, that sounds fine to me (although it would be against the style guide).

Reply:
@jasmeet0817 I would add an `enum` and use `Map<_MyThresholdEnum, Color>` in order to clarify things for your points 1 and 2 (I agree with both).
I don't agree with point 3 though.
And I still find problematic the `dynamic`/cast algo.
Anyway, the idea would either to keep methods without a class (the way you coded), or class instances (like I suggested). Static methods only would be against the dart best practice you were mentioning.

If another reviewer approves your code as it is then so be it.

Reply:
Thanks again @monsieurtanuki, I think the "dynamic/cast" bit being problematic is a good point, I'm unfamiliar with Dart so I don't know exactly, I thought it should be alright because the public methods always return non-dynamic variables so any library user will see the functions as strongly typed.

But it's good feedback, I'll check with someone from the Flutter team for advice.